servers/services/src/main/java/org/projectnessie/services/impl/TreeApiImplWithAuthorization.java

```diff
@@ -131,13 +131,11 @@ protected void assignReference(NamedRef ref, String expectedHash, Reference assi
   @Override
   protected void deleteReference(NamedRef ref, String expectedHash)
       throws NessieConflictException, NessieNotFoundException {
-    BatchAccessChecker check = startAccessCheck();
     if (ref instanceof BranchName && getConfig().getDefaultBranch().equals(ref.getName())) {
-      check.canDeleteDefaultBranch();
-    } else {
-      check.canDeleteReference(ref);
+      throw new IllegalArgumentException(
+          "Default branch '" + ref.getName() + "' cannot be deleted.");
```

Comment:
I do not really have a strong opinion on forbidding deleting the default branch, but from a general usability perspective I guess it might be worth reversing the default on `canDeleteReference` to `false` so that we still allow full functionality at the API level, but the admin has to enabled deleting the default branch explicitly. WDYT?

Reply:
That was the existing functionality, that we discussed in the ticket to change:
https://github.com/projectnessie/nessie/issues/3401